README.md

```diff
@@ -223,12 +223,18 @@ Instead, it relies on LVM2 to lock around operations that are not reentrant.
 
 #### Logical volume naming
 
-The volume group name is specified at startup through the
-`-volume-group` argument.
+The volume group name is specified at startup through the `-volume-group` argument.
 
-Logical volumes are named according to the following pattern
-`<volume-group-name>_<logical-volume-name>`.
+Logical volumes names are chosen from randomly-generated, base36-encoded numbers.
+The CO-specified volume name is captured in a LV tag: `<prefix>_<fn(CO-specified-name)>`, where `<prefix>` is either of:
 
+* `VN.` - used when the CO-specified name contains *only* characters safe for LVM tags; `fn` is the identity func here.
+* `VN+` - used when the CO-specified name contains 1 or more characters unsafe for LVM tags; `fn` is base64-rawurl-encoding here.
```

Comment:
I was wondering if there's a "filename-safe" version (or an encoding lib w/ custom allowed char set and escape char) of quoted-printable encoding (https://en.wikipedia.org/wiki/Quoted-printable) so we can unify these two cases here.

Reply:
I don't think quoted-printable works because it allows spaces, and probably other tag-unsafe chars. The alphabet for base64-rawurl-encoding is a (slightly) smaller subset of the allowed chars for LVM tags and seems like a perfect fit. I don't think the plugin should jump through hoops for CO's that generate names with unsafe chars, though I do get your point about unifying the cases. Having separate prefixes is not ideal, but I wanted to optimize for COs that play by the rules so that the ops experience is better there.

Reply:
Yeah I know that the quoted-printable encoding won't work. I was wondering if there's any golang lib that allows us to specify the allowed/disallowed chars and the escape char, and to a quoted-printable-like encoding. If yes then we can consider that, if no than sticking w/ what proposed here is not too bad.

Reply:
The allowed chars for LVM tags are pretty limited. That said:

* the std base64 package lets you define a custom alphabet: https://golang.org/pkg/encoding/base64/#NewEncoding
* the std quotedprintable library doesn't support custom alphabets, or (seemingly) "no-padding".

It sounds like you're pushing for the custom-alpha-quoted-printable because the generated tags would bear a closer resemblance to the CO-specified volume name. It's a nice idea. I'll poke around a bit and see what I can dig up.

Reply:
We could probably write a variant of a URL query encoder that was slightly more restrictive w/ respect to allowed chars. Is it really worth the effort to (a) write the encoder; (b) test the encoder; (c) properly document the encoder (so that Dan/whomever can write a decoder); (d) maintain all the above?

Reply:
Said another way, I think there's real value in using a pre-existing encoding scheme that (a) someone else wrote; (b) there's already a decoder for (that someone else wrote); (c) it's thoroughly documented and unlikely to change any time soon; (d) someone else maintains it all; (e) it's well understood.